src/EnergyPlus/PollutionModule.cc

```diff
@@ -2188,3998 +419,214 @@ void SetupPollutionMeterReporting(EnergyPlusData &state)
+        for (int iPollFuel = 0; iPollFuel < (int)PollFuel::Num; ++iPollFuel) {   
+            auto &pollCoeff = pm->pollCoeffs[iPollFuel];
+            auto &pollComp = pm->pollComps[(int)pollFuel2pollFuelComponent[iPollFuel]];
+
+            if (pollCoeff.used) {
+                pollComp.pollutantVals[iPoll] = 0.0;
+                Real64 pollutantVal = pollCoeff.pollutantCoeffs[iPoll] * 0.001;
+                if (pollCoeff.pollutantSchedNums[iPoll] != 0) {
+                    pollutantVal *= ScheduleManager::GetCurrentScheduleValue(state, pollCoeff.pollutantSchedNums[iPoll]);
+                }
+                pollComp.pollutantVals[iPoll] =
+                    pm->facilityMeterFuelComponentVals[(int)pollFuel2pollFuelComponent[iPollFuel]] * 1.0e-6 * pollutantVal;
+            }
+
+            pm->pollutantVals[iPoll] += pollComp.pollutantVals[iPoll];
+        } // for (iPollFactor)
+    } // for (iPoll)
+
+    auto const &pollCoeffElec = pm->pollCoeffs[(int)PollFuel::Electricity];
+    auto &pollCompElec = pm->pollComps[(int)PollFuelComponent::Electricity];
+    auto &pollCompElecPurchased = pm->pollComps[(int)PollFuelComponent::ElectricityPurchased];
+    auto &pollCompElecSurplusSold = pm->pollComps[(int)PollFuelComponent::ElectricitySurplusSold];
+    
+    pollCompElec.sourceVal = pm->facilityMeterFuelComponentVals[(int)PollFuelComponent::Electricity] * pollCoeffElec.sourceCoeff;
+    pollCompElecPurchased.sourceVal = pm->facilityMeterFuelComponentVals[(int)PollFuelComponent::ElectricityPurchased] * pollCoeffElec.sourceCoeff;
+    pollCompElecSurplusSold.sourceVal = pm->facilityMeterFuelComponentVals[(int)PollFuelComponent::ElectricitySurplusSold] * pollCoeffElec.sourceCoeff;
+
+    if (pollCoeffElec.sourceSchedNum != 0) {
+        Real64 pollCoeffElecSchedVal = ScheduleManager::GetCurrentScheduleValue(state, pollCoeffElec.sourceSchedNum);
+        pollCompElec.sourceVal *= pollCoeffElecSchedVal;
+        pollCompElecPurchased.sourceVal *= pollCoeffElecSchedVal;
+        pollCompElecSurplusSold.sourceVal *= pollCoeffElecSchedVal;
+    }
+    
+    // does not include district heating or steam
+    auto const &pollCoeff = pm->pollCoeffs[(int)PollFuel::NaturalGas];
+    auto &pollComp = pm->pollComps[(int)PollFuelComponent::NaturalGas];
+    pollComp.sourceVal = pm->facilityMeterVals[(int)PollFacilityMeter::NaturalGas] * pollCoeff.sourceCoeff;
+    if (pollCoeff.sourceSchedNum != 0) {
+         pollComp.sourceVal *= ScheduleManager::GetCurrentScheduleValue(state, pollCoeff.sourceSchedNum);
+    }
+
+    for (PollFuel pollFuel : {PollFuel::FuelOil1, PollFuel::FuelOil2, PollFuel::Diesel, PollFuel::Gasoline,
+                              PollFuel::Propane, PollFuel::Coal, PollFuel::OtherFuel1, PollFuel::OtherFuel2}) {
+        auto const &pollCoeff = pm->pollCoeffs[(int)pollFuel];
+        PollFuelComponent pollFuelComponent = pollFuel2pollFuelComponent[(int)pollFuel];
+        auto &pollComp = pm->pollComps[(int)pollFuelComponent];
```

Comment:
[src/EnergyPlus/PollutionModule.cc:620]:(style),[shadowVariable],Local variable 'pollCoeff' shadows outer variable
[src/EnergyPlus/PollutionModule.cc:622]:(style),[shadowVariable],Local variable 'pollComp' shadows outer variable

Reply:
Addressed this.